src/python/m5/SimObject.py

```diff
@@ -1236,6 +1248,14 @@ def getCCParams(self):
                 else:
                     for v in value:
                         getattr(cc_params, param).append(v)
+            elif isinstance(self._params[param], DictParamDesc):
+                assert isinstance(value, dict)
+                dic = getattr(cc_params, param)
+                assert not len(dic)
```

Comment:
`dic` should be a `dict`. So, I think that `assert not dic, "Dictionary param should not be empty"` would be more clear.

Reply:
Ah, sorry. I see now. A more descriptive error could potentially be useful for future developers, but since it's not user-facing this is just a nit.

Reply:
Done!